sdk/storage/azure-storage-blob/azure/storage/blob/blob_client.py

```diff
@@ -2487,3 +2487,164 @@ def append_block( # type: ignore
+
+        append_conditions = None
+        if maxsize_condition or appendpos_condition is not None:
+            append_conditions = AppendPositionAccessConditions(
+                max_size=maxsize_condition,
+                append_position=appendpos_condition
+            )
+        access_conditions = get_access_conditions(kwargs.pop('lease', None))
+        mod_conditions = ModifiedAccessConditions(
+            if_modified_since=kwargs.pop('if_modified_since', None),
+            if_unmodified_since=kwargs.pop('if_unmodified_since', None),
+            if_match=kwargs.pop('if_match', None),
+            if_none_match=kwargs.pop('if_none_match', None))
+        source_mod_conditions = SourceModifiedAccessConditions(
+            source_if_modified_since=kwargs.pop('source_if_modified_since', None),
+            source_if_unmodified_since=kwargs.pop('source_if_unmodified_since', None),
+            source_if_match=kwargs.pop('source_if_match', None),
+            source_if_none_match=kwargs.pop('source_if_none_match', None))
+
+        options = {
+            'source_url': copy_source_url,
+            'content_length': 0,
+            'source_range': source_range,
+            'source_content_md5': source_content_md5,
+            'timeout': kwargs.pop('timeout', None),
+            'transactional_content_md5': None,
+            'lease_access_conditions': access_conditions,
+            'append_position_access_conditions': append_conditions,
+            'modified_access_conditions': mod_conditions,
+            'source_modified_access_conditions': source_mod_conditions,
+            'cls': return_response_headers}
+        options.update(kwargs)
+        return options
+
+    @distributed_trace
+    def append_block_from_url(self, copy_source_url,  # type: str
+                              source_range_start=None,  # type Optional[int]
+                              source_range_end=None,  # type Optional[int]
+                              source_content_md5=None,  # type: Optional[bytearray]
+                              maxsize_condition=None,  # type: Optional[int]
+                              appendpos_condition=None,  # type: Optional[int]
+                              lease=None, if_modified_since=None,
+                              if_unmodified_since=None, if_match=None,
+                              if_none_match=None, source_if_modified_since=None,
+                              source_if_unmodified_since=None, source_if_match=None,
+                              source_if_none_match=None, timeout=None):
+        # type: (...) -> Dict[str, Union[str, datetime, int]]
+        """
+        Creates a new block to be committed as part of a blob, where the contents are read from a source url.
+
+        :param str copy_source_url:
+            The URL of the source data. It can point to any Azure Blob or File, that is either public or has a
+            shared access signature attached.
```

Comment:
I believe external sources (not inside Azure) are supported too, could you please verify?

Reply:
It doesn't work for external url, got this error in response **The requested URI does not represent any resource on the server.**

Reply:
Ah I see, sounds good